enterprise/server/oci/ocifetcher/ocifetcher.go

```diff
@@ -42,6 +49,32 @@ func NewClient(allowedPrivateIPs []*net.IPNet, mirrors []interfaces.MirrorConfig
 	}
 }
 
+func RegisterClient(env *real_environment.RealEnv) error {
+	allowedPrivateIPNets, err := ParseAllowedPrivateIPs()
+	if err != nil {
+		return err
+	}
+	client := NewClient(allowedPrivateIPNets, *mirrors)
```

Comment:
Checking my understanding: seems like the client should not know about the allowed private registry IPs (only the cache proxy / apps which actually interact with the registry should care about this)? I guess this is just a side effect of the partially-migrated state we're in, where the client and "server" are sort of entangled / local to the executor? (If that's right, might make sense to drop a note/TODO re. moving this to the server)

Reply:
Right now, I'm moving the fetching logic from the `oci` package to the `OCIFetcherClient`.

The end state will be the fetching logic lives on "the server" (be it the app or enterprise cache proxies).
For now, I'm moving functionality behind the OCIFetcherClient interface before implementing the OCIFetcherServer.
I'll then gradually move executors from using a "local" client to using a thin client that actually makes gRPC calls to a remote service.

Great point for a TODO!